term/src/terminalstate/image.rs

```diff
@@ -115,6 +131,15 @@ impl TerminalState {
             .context("computing xpos")?;
 
         let cursor_x = self.cursor.x;
+
+        let width_in_cells = fullcells_width + (remainder_width_cell > 0) as usize;
+        let height_in_cells = fullcells_height + (remainder_height_cell > 0) as usize;
```

Comment:
stylistically, I think these conversions from bool to integer are a bit hard to follow; they are certainly succinct, but it's not super clear what is happening, or perhaps more important, whether that what happens is intentional.

I propose adding a little helper function, something like:

```rust
use num_traits::{One, Zero};

/// Returns `1` if `b` is true, else `0`,
fn one_or_zero<T: Zero+One>(b: bool) -> T{
    if b {
        T::one()
    } else {
        T::zero()
    }
}
```

then the logic here could be:

```suggestion
        let width_in_cells = fullcells_width + one_or_zero(remainder_width_cell > 0);
        let height_in_cells = fullcells_height + one_or_zero(remainder_height_cell > 0);
```